### ISSUE TITLE
Sunfish hangs before making final move

### ISSUE THREAD

**Issue by thomasahle** [thomasahle/sunfish#63]
In the following mate-in-one, sunfish does the first 17 plies very fast, and then has a very long think before ply 18. This makes it run out of time and makes cutechess think its connection has sttalled. 
```
setboard 8/8/1Q4B1/8/5P2/P1k2P1P/6P1/6K1 w - - 1 43
post
go
  1     1822        0            43 	b6e6
  2    69290        1           141 	b6b4
  3    69290        1           162 	b6b4
  4    69290        1           181 	b6b4
  5    69290        1           200 	b6b4
  6    69290        2           283 	b6b4
  7    69290        2           402 	b6b4
  8    69290        3           509 	b6b4
  9    69290        3           616 	b6b4
 10    69290        4           872 	b6b4
 11    69290        5          1172 	b6b4
 12    69290        6          1462 	b6b4
 13    69290        7          1752 	b6b4
 14    69290        8          2260 	b6b4
 15    69290       10          2848 	b6b4
 16    69290       12          3416 	b6b4
 17    69290       14          3984 	b6b4
 18    69290     6236       1855272 	b6b4
move b6b4
```

**Comment by thomasahle** [comment]
Another example is
```
time 4669
otim 5465
setboard 6Q1/5R2/8/7k/6p1/8/6PP/6K1 w - - 3 47
post
go
  1     1721        0            37 	h2h3
  2    69290        0            74 	f7h7
  3    69290        0            89 	f7h7
  4    69290        0           102 	f7h7
...

**Comment by thomasahle** [comment]
A few more positions for testing
```
8/2N5/8/1P6/1R6/P4R2/5P1P/2k1K3 w - - 5 44
8/7P/6R1/8/7k/1P6/8/6K1 w - - 5 48
8/5R2/4Q3/1k6/p1p5/P7/1P6/6K1 w - - 0 56
6Q1/8/8/8/1P5p/7k/5K2/8 w - - 0 50
8/8/1Q4B1/8/5P2/P1k2P1P/6P1/6K1 w - - 1 43
6Q1/8/7k/6p1/6P1/4N3/8/5RK1 w - - 3 61
```

**Comment by thomasahle** [comment]
Fixed by commit 386f578.
